diff --git a/node_modules/react-native-test-app/android/app/src/main/AndroidManifest.xml b/node_modules/react-native-test-app/android/app/src/main/AndroidManifest.xml
index 444ed05..9f9055c 100644
--- a/node_modules/react-native-test-app/android/app/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-test-app/android/app/src/main/AndroidManifest.xml
@@ -20,16 +20,20 @@
         android:theme="@style/AppTheme"
         android:usesCleartextTraffic="true"
         tools:targetApi="m"
-    >
+      >
         <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
 
-        <activity android:name="com.microsoft.reacttestapp.MainActivity" android:exported="true">
+        <activity android:name="com.microsoft.reacttestapp.MainActivity" android:exported="true"
+          android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
+        >
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
 
-        <activity android:name="com.microsoft.reacttestapp.component.ComponentActivity" />
+        <activity android:name="com.microsoft.reacttestapp.component.ComponentActivity"
+          android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
+        />
     </application>
 </manifest>
diff --git a/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/MainActivity.kt b/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/MainActivity.kt
index a421e4b..318264f 100644
--- a/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/MainActivity.kt
+++ b/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/MainActivity.kt
@@ -25,6 +25,7 @@ import com.microsoft.reacttestapp.component.ComponentViewModel
 import com.microsoft.reacttestapp.manifest.Component
 import com.microsoft.reacttestapp.react.AppRegistry
 import com.microsoft.reacttestapp.react.BundleSource
+import eu.reactnativetraining.ThemeControlModule
 
 class MainActivity : ReactActivity() {
 
@@ -66,7 +67,8 @@ class MainActivity : ReactActivity() {
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
+        super.onCreate(null)
+        ThemeControlModule.Companion.recoverApplicationTheme(this)
 
         val (manifest, checksum) = testApp.manifestProvider.fromResources()
         val components = manifest.components ?: listOf()
diff --git a/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/component/ComponentActivity.kt b/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/component/ComponentActivity.kt
index a7ccf92..611e1fb 100644
--- a/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/component/ComponentActivity.kt
+++ b/node_modules/react-native-test-app/android/app/src/main/java/com/microsoft/reacttestapp/component/ComponentActivity.kt
@@ -8,6 +8,7 @@ import androidx.fragment.app.Fragment
 import com.facebook.react.ReactActivity
 import com.facebook.react.ReactActivityDelegate
 import com.microsoft.reacttestapp.BuildConfig
+import eu.reactnativetraining.ThemeControlModule
 
 class ComponentActivity : ReactActivity() {
 
@@ -35,7 +36,8 @@ class ComponentActivity : ReactActivity() {
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
+        super.onCreate(null)
+        ThemeControlModule.Companion.recoverApplicationTheme(this)
 
         @Suppress("SENSELESS_COMPARISON")
         if (BuildConfig.ReactTestApp_singleApp === null) {
